skyportal/handlers/api/photometry.py

```diff
@@ -674,61 +676,49 @@ def put(self):
+                # save existing ids:
+                existing_ids.extend([d.id for d in duplicated_photometry])
+
+                # update groups if necessary
+                for duplicate in duplicated_photometry:
+                    duplicate_group_ids = set([g.id for g in duplicate.groups])
+
+                    # posting to new groups?
+                    if len(set(group_ids) - duplicate_group_ids) > 0:
+                        # select old + new groups
+                        group_ids_update = set(group_ids).union(duplicate_group_ids)
+                        groups = (
+                            DBSession()
+                            .query(Group)
+                            .filter(Group.id.in_(group_ids_update))
+                            .all()
                         )
-                        # posting to new groups?
-                        if len(set(group_ids) - row_group_ids) > 0:
-                            # select old + new groups
-                            group_ids_update = set(group_ids).union(row_group_ids)
-                            groups = (
-                                DBSession()
-                                .query(Group)
-                                .filter(Group.id.in_(group_ids_update))
-                                .all()
-                            )
-                            # update the corresponding photometry entry in the db
-                            index_to_update = np.where(w)[0][0]
-                            existing_photometry[index_to_update].groups = groups
-
-                            updated_photometry_groups = True
-
-                    # save existing ids:
-                    w_ids = df_existing_photometry.hash.isin(df.hash)
-                    existing_ids.extend(
-                        df_existing_photometry.loc[w_ids, "id"].values.tolist()
-                    )
+                        # update the corresponding photometry entry in the db
+                        duplicate.groups = groups
 
-                    # now safely drop the duplicates:
-                    df = df.drop(index=np.where(w_oid_duplicate)[0])
+                        updated_photometry_groups = True
+
+                w_oid_duplicate = (df.obj_id == oid) & (df.hash.isin(existing_hashes))
+                # now safely drop the duplicates:
+                df = df.drop(index=np.where(w_oid_duplicate)[0])
 
         # posted data contains only duplicates already present in the db?
         if len(df) == 0:
             if updated_photometry_groups:
                 DBSession().commit()
-                return self.success(data={"ids": existing_ids})
-            else:
-                return self.success(data={"ids": existing_ids})
+            return self.success(data={"ids": existing_ids})
 
         # pre-fetch the photometry PKs. these are not guaranteed to be
         # gapless (e.g., 1, 2, 3, 4, 5, ...) but they are guaranteed
```

Comment:
Only execute this section if there is new photometry to be inserted. Then do one single commit and return statement at the end of the handler